tiled/src/main/kotlin/ktx/tiled/tiledMap.kt

```diff
@@ -0,0 +1,99 @@
+package ktx.tiled
+
+import com.badlogic.gdx.maps.MapLayer
+import com.badlogic.gdx.maps.MapObject
+import com.badlogic.gdx.maps.MapProperties
+import com.badlogic.gdx.maps.tiled.TiledMap
+
+private val DEFAULT_MAP_LAYER = MapLayer().apply { name = "ktx-default-map-layer" }
+
+/**
+ * Extension method to directly access the [MapProperties] of a [TiledMap]. The type is automatically
+ * derived from the type of the given default value. If the property is not defined the defaultValue will be returned.
+ * @param key property name
+ * @param defaultValue default value in case the property is missing
+ * @return value of the property or defaultValue if property is missing
+ */
+inline fun <reified T> TiledMap.property(key: String, defaultValue: T): T = this.properties[key, defaultValue, T::class.java]
+
+/**
+ * Extension method to directly access the [MapProperties] of a [TiledMap]. If the property
+ * is not defined then this method returns null.
+ * @param key property name
+ * @return value of the property or null if the property is missing
+ */
+inline fun <reified T> TiledMap.propertyOrNull(key: String): T? = this.properties[key, T::class.java]
+
+/**
+ * Extension method to directly access the [MapProperties] of a [TiledMap] and its [containsKey][MapProperties.containsKey] method
+ * @param key property name
+ * @return true if the property exists. Otherwise false
+ */
+fun TiledMap.containsProperty(key: String) = properties.containsKey(key)
+
+/**
+ * Extension property to retrieve the width of the [TiledMap]. If the map does not have
+ * a **width** property then 0 is returned
+ */
+val TiledMap.width: Int
+    get() = property("width", 0)
+
+/**
+ * Extension property to retrieve the height of the [TiledMap]. If the map does not have
+ * a **height** property then 0 is returned
+ */
+val TiledMap.height: Int
+    get() = property("height", 0)
+
+/**
+ * Extension property to retrieve the tile width of each tile of the [TiledMap]. If the map does not have
+ * a **tilewidth** property then 0 is returned
+ */
+val TiledMap.tileWidth: Int
+    get() = property("tilewidth", 0)
+
+/**
+ * Extension property to retrieve the tile height of each tile of the [TiledMap]. If the map does not have
+ * a **tileheight** property then 0 is returned
+ */
+val TiledMap.tileHeight: Int
+    get() = property("tileheight", 0)
```

Comment:
Same comment about defaults applies here. I think I'd rather have to handle an exception rather than get an unexpected 0 on tile height, width, etc.